src/Adapter/MSTest.TestAdapter/MSTestSettings.cs

```diff
@@ -253,7 +257,61 @@ public static void PopulateSettings(MSTestSettings settings)
     /// The discovery context that contains the runsettings.
     /// </param>
     [Obsolete("this function will be removed in v4.0.0")]
-    public static void PopulateSettings(IDiscoveryContext? context) => PopulateSettings(context, null);
+    public static void PopulateSettings(IDiscoveryContext? context) => PopulateSettings(context, null, null);
+
+    private static bool IsRunSettingsFileHesMSTestSettings(string? runSettingsXml) => IsRunSettingsFileHasSettingName(runSettingsXml, SettingsName, out _) || IsRunSettingsFileHasSettingName(runSettingsXml, SettingsNameAlias, out _)
+                    || (IsRunSettingsFileHasSettingName(runSettingsXml, ConfigurationName, out XmlReader? runConfigrationReader) && HasRelevantKeys(runConfigrationReader));
+
+    private static bool HasRelevantKeys(XmlReader? reader)
```

Comment:
This seems to only have sense in the cotnext above so I would make it local function of the previous or rename it to something more menaingfull